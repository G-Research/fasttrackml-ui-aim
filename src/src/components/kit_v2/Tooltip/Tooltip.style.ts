import * as Tooltip from '@radix-ui/react-tooltip';

import { keyframes, styled } from 'config/stitches';

const slideUpAndFade = keyframes({
  '0%': { opacity: 0, transform: 'translateY(2px)' },
  '100%': { opacity: 1, transform: 'translateY(0)' },
});

const slideRightAndFade = keyframes({
  '0%': { opacity: 0, transform: 'translateX(-2px)' },
  '100%': { opacity: 1, transform: 'translateX(0)' },
});

const slideDownAndFade = keyframes({
  '0%': { opacity: 0, transform: 'translateY(-2px)' },
  '100%': { opacity: 1, transform: 'translateY(0)' },
});

const slideLeftAndFade = keyframes({
  '0%': { opacity: 0, transform: 'translateX(2px)' },
  '100%': { opacity: 1, transform: 'translateX(0)' },
});

const TooltipContent = styled(Tooltip.Content, {
  borderRadius: 4,
  padding: '10px 15px',
  fontSize: 15,
  lineHeight: 1,
  backgroundColor: 'white',
  boxShadow:
    'hsl(206 22% 7% / 35%) 0px 10px 38px -10px, hsl(206 22% 7% / 20%) 0px 10px 20px -15px',
  userSelect: 'none',
  animationDuration: '400ms',
  animationTimingFunction: 'cubic-bezier(0.16, 1, 0.3, 1)',
  willChange: 'transform, opacity',
  '&[data-state="delayed-open"]': {
    '&[data-side="top"]': { animationName: slideDownAndFade },
    '&[data-side="right"]': { animationName: slideLeftAndFade },
    '&[data-side="bottom"]': { animationName: slideUpAndFade },
    '&[data-side="left"]': { animationName: slideRightAndFade },
  },
});

const TooltipArrow = styled(Tooltip.Arrow, {
  fill: 'white',
});

export { TooltipContent, TooltipArrow };
