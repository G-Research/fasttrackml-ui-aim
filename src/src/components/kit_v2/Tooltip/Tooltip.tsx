import React from 'react';

import * as TooltipPrimitive from '@radix-ui/react-tooltip';

import { TooltipArrow, TooltipContent } from './Tooltip.style';
import { ITooltipProps } from './Tooltip.d';

/**
 * @description Tooltip component is for displaying a tooltip
 * Tooltip component params
 * @param {React.ReactNode} children - React children
 * @param {string} content - Tooltip content
 * @param {number} delayDuration - Delay duration
 * @param {boolean} disableHoverableContent - Disable hoverable content
 * @param {number} skipDelayDuration - Skip delay duration
 * @param {React.ReactNode} contentProps - Content props
 * @returns {React.FunctionComponentElement<React.ReactNode>}
 * @constructor
 * @example
 * <Tooltip
 *  content='Tooltip content'
 *  delayDuration={700}
 *  disableHoverableContent={false}
 *  skipDelayDuration={300}
 * >
 * <Button>Open tooltip</Button>
 * </Tooltip>
 */
function Tooltip({
  content,
  delayDuration = 700,
  disableHoverableContent = false,
  skipDelayDuration = 300,
  contentProps = {},
  children,
  ...props
}: ITooltipProps): React.FunctionComponentElement<React.ReactNode> {
  return (
    <TooltipPrimitive.Provider
      delayDuration={delayDuration}
      disableHoverableContent={disableHoverableContent}
      skipDelayDuration={skipDelayDuration}
    >
      <TooltipPrimitive.Root {...props}>
        <TooltipPrimitive.Trigger asChild>{children}</TooltipPrimitive.Trigger>
        <TooltipPrimitive.Portal>
          <TooltipContent
            data-testid='tooltip-content'
            sideOffset={5}
            {...contentProps}
          >
            {content}
            <TooltipArrow />
          </TooltipContent>
        </TooltipPrimitive.Portal>
      </TooltipPrimitive.Root>
    </TooltipPrimitive.Provider>
  );
}

Tooltip.displayName = 'Tooltip';
export default React.memo(Tooltip);
