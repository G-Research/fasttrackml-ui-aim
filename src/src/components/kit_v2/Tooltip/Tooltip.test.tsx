import { render } from '@testing-library/react';

import Tooltip from './Tooltip';

// Test of Tooltip component
describe('<Tooltip content="Tooltip">Hover</Tooltip>', () => {
  test('Renders Correctly', () => {
    const { asFragment } = render(<Tooltip content='Tooltip'>Hover</Tooltip>);
    expect(asFragment()).toMatchSnapshot();
  });
});
