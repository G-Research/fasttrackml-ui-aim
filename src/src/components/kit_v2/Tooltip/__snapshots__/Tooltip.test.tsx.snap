// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip content="Tooltip">Hover</Tooltip> Renders Correctly 1`] = `<DocumentFragment />`;
